### PR TITLE
update the staled controller name to projectcontour.io/gateway-controller

### PR DESCRIPTION
**Pull request by izturn**
Signed-off-by: Gang Liu [gang.liu@daocloud.io](mailto:gang.liu@daocloud.io)

Update the staled controller name to `projectcontour.io/gateway-controller` to fix `examples/gateway` and more

### CODE REVIEW

File: design/configuration-crd.md [hunk 1]
```diff
@@ -107,7 +107,7 @@ spec:
     cluster:
       dnsLookupFamily: auto
   gateway:
-      controllerName: projectcontour.io/projectcontour/contour
+      controllerName: projectcontour.io/gateway-controller
```
Comment (sunjayBhatia) [inline design/configuration-crd.md:110]: i would say we can maybe leave the `design` directory as-is, so these documents capture what the thinking was when they were written, if they need a deliberate update we should do so more explicitly

Comment (skriss) [reply design/configuration-crd.md:110]: +1

**Review by skriss — CHANGES_REQUESTED** [review]
Thanks for the PR @izturn. We should also update the following documentation pages with similar changes:

- https://github.com/projectcontour/contour/blob/main/site/content/guides/gateway-api.md
- https://github.com/projectcontour/contour/blob/main/site/content/docs/main/configuration.md

**Comment by skriss** [comment]
Also please add a `-docs.md` changelog, I think this is worth calling out as a change.

File: design/configuration-crd.md (hunk 1 above)
Comment (izturn) [reply design/configuration-crd.md]: done

File: changelogs/unreleased/4966-izturn-docs.md [hunk 2]
```diff
@@ -0,0 +1 @@
+Change the default controller name from `projectcontour.io/projectcontour/contour` to `projectcontour.io/gateway-controller`
```
Comment (skriss) [inline changelogs/unreleased/4966-izturn-docs.md:1]:
```suggestion
Gateway API: change the default controller name from `projectcontour.io/projectcontour/contour` to `projectcontour.io/gateway-controller` for static provisioning.
```

**Review by skriss — CHANGES_REQUESTED** [review]

File: site/content/guides/gateway-api.md [hunk 3]
```diff
@@ -121,7 +121,7 @@ metadata:
 data:
   contour.yaml: |
     gateway:
-      controllerName: projectcontour.io/projectcontour/contour
+      controllerName:projectcontour.io/gateway-controller
```
[the remark this answers is not in the document]
Comment (skriss) [reply site/content/guides/gateway-api.md:124]: Still missing the space on this one.

**Comment by izturn** [comment]
@skriss  PTAL

**Review by skriss — APPROVED** [review]

**Pull request merged**